tests/unittests/GraphOptzTest.cpp

```diff
@@ -1817,6 +1825,86 @@ TEST_F(GraphOptz, foldQuantizeIntoVarMultipleUsages) {
   }
 }
 
+static SaveNode *getUniqueSaveNode(Function *F) {
+  SaveNode *foundSaveNode = nullptr;
+  for (auto &node : F->getNodes()) {
+    if (auto *s = llvm::dyn_cast<SaveNode>(&node)) {
+      EXPECT_EQ(foundSaveNode, nullptr);
+      foundSaveNode = s;
+    }
+  }
+  EXPECT_NE(foundSaveNode, nullptr);
+  return foundSaveNode;
+}
+
+/// Mock backend that requests the pre-quantization of constants.
+class MockBackendPrequantizeConst : public MockBackend {
+  bool shouldPreQuantizeConstants() const override { return true; }
+  bool isOpSupported(const NodeInfo &) const override { return true; }
+  bool transformPostLowering(Function *F, CompilationContext &,
+                             const glow::runtime::DeviceInfo *) const override {
+    // Check the IR.
+    EXPECT_EQ(F->getNodes().size(), 1);
+    auto *save = getUniqueSaveNode(F);
+    EXPECT_TRUE(llvm::isa<Constant>(save->getInput()));
+
+    return false;
+  }
+};
+/// Mock backend that requests the non pre-quantization of constants.
+class MockBackendNotPrequantizeConst : public MockBackend {
+  bool shouldPreQuantizeConstants() const override { return false; }
+  bool isOpSupported(const NodeInfo &) const override { return true; }
+  bool transformPostLowering(Function *F, CompilationContext &,
+                             const glow::runtime::DeviceInfo *) const override {
+    // Check the IR.
+    EXPECT_EQ(F->getNodes().size(), 2);
+    auto *save = getUniqueSaveNode(F);
+    auto *quant = llvm::dyn_cast<QuantizeNode>(save->getInput());
+    EXPECT_TRUE(quant);
+    EXPECT_TRUE(llvm::isa<Constant>(quant->getInput()));
+
+    return false;
+  }
+};
+
+/// Check the backend actual constant quantization is done before post-lowering.
+TEST_F(GraphOptz, foldQuantizeIntoConstantBeforePostLowering) {
+  // auto *input = mod_.createPlaceholder(ElemKind::FloatTy, {4}, "input",
+  // true); *bindings_.allocate(input) = {10, 10, 10, 10};
+
+  auto *input = mod_.createConstant(ElemKind::FloatTy, {4}, "input");
+  input->getHandle<float>() = {10, 10, 10, 10};
+  auto qType = mod_.uniqueType(ElemKind::Int8QTy, {4}, 2, 0);
+  auto *Q = F_->createQuantize("quantize", input, qType);
+  F_->createSave("save", Q);
+
+  CompilationContext cctx;
+  BackendOptions opts;
+  auto B = MockBackendPrequantizeConst();
+  EXIT_ON_ERR(::glow::optimizeFunction(F_, B, cctx));
+}
+
+/// Check the backend actual constant quantization is done after post-lowering.
+TEST_F(GraphOptz, foldQuantizeIntoConstantAfterPostLowering) {
```

Comment:
Actually, this test should be identical to `foldQuantizeIntoConstantBeforePostLowering` except for the different backend, right? Can we combine these into a single helper function templatized on which mock backend to use?

Reply:
OK, it means that I then need to add the test you were requested in the previous feedback :)
There is then a good reason to do it now.

Reply:
Done.